src/main/src/main/java/org/geoserver/catalog/FeatureTypeInfo.java

```diff
@@ -193,4 +193,13 @@
 	
 	void setCircularArcPresent(boolean arcsPresent);
 
+
+	/**
+	 * Whether querying the full number of features is banned for this feature type.
+	 * 
+	 * @return True if a full feature count should not be attempted, false otherwise
+	 */
+	boolean isTotalCountBanned();
+
+	void setTotalCountBanned(boolean totalCountBanned);
```

Comment:
Banning is a strong word and selecting the right word is a bit problematic: using this feature potentially breaks the query interface (knowingly) as the count is not done at all, but on the other hand, because of multi-query GetFeature calls, this "ban" cannot be enforced at all times.

Would totalCountOmittedWhenPossible work for you, or is it too long and unwieldy?